be1-go/hub/inbox.go

```diff
@@ -75,3 +114,73 @@ func (i *inbox) getMessage(messageID []byte) (*message.Message, bool) {
 	}
 	return msgInfo.message, true
 }
+
+func (i *inbox) storeMessageInDB(messageInfo *messageInfo) error {
+	db, err := sql.Open("sqlite3", os.Getenv("HUB_DB"))
+	if err != nil {
+		return xerrors.Errorf("failed to open connection: %v", err)
+	}
+
+	defer db.Close()
+
+	query := `
+		INSERT INTO 
+			message_info(
+				message_id, 
+				sender, 
+				message_signature, 
+				raw_data, 
+				message_timestamp, 
+				lao_channel_id) 
+		VALUES(?, ?, ?, ?, ?, ?)`
+
+	stmt, err := db.Prepare(query)
+	if err != nil {
+		return xerrors.Errorf("failed to prepare query: %v", err)
+	}
+
+	defer stmt.Close()
+
+	msg := messageInfo.message
+
+	_, err = stmt.Exec(msg.MessageID, msg.Sender, msg.Signature, msg.RawData, messageInfo.storedTime, i.channelID)
```

Comment:
I'm not sure if the underlying []byte will be converted to base64url encoded string representations here automatically. Did you try to observe the rows here in a viewer?

Reply:
MessageID, Sender, and Signature are saved as binary. I tried saving/loading and it shows that it can successfully restore back the data to the correct format.